acctest/plugin/component_acc_test.go

```diff
@@ -32,13 +31,6 @@ func TestAccInitAndBuildBasicAmazonAmiDatasource(t *testing.T) {
 		Setup: func() error {
 			return cleanupPluginInstallation(plugin)
 		},
-		Teardown: func() error {
-			helper := amazonacc.AMIHelper{
```

Comment:
> Related to this though, is there a reason why those tests live here if they rely on the AWS plugin? They could be moved over there

This is a good question and not one that I've looked into just yet. I think they were added as a way to ensure datasource and multi-plugins continue working in Packer core. 

> I understand keeping this means we still partially embed the plugin (though not in the packer binary), but I would be careful removing this, as this will clutter the AWS account with unused AMIs.

So this helper is not needed if we use `skip_create_ami` in the template. Which is what I did. We can make the argument that we want to test the ami creation part too but these tests don't do anything fancy, as they just validate the plugin works. The Amazon tests should validate this for us. So I removed the helper for clean up and updated the config to not generate the AMI to avoid the clutter.

Reply:
Agreed, the AMI creation part is on the Amazon plugin, not Packer core. If the aim is to have more confidence that Packer can load plugins and run their components, we can keep them in here, if we do `skip_create_ami`, that's good enough to me, we won't clutter the account, and it succeeds in testing what it needs to.